core/services/ocr2/delegate.go

```diff
@@ -473,6 +479,147 @@ func GetEVMEffectiveTransmitterID(jb *job.Job, chain evm.Chain, lggr logger.Suga
 	return spec.TransmitterID.String, nil
 }
 
+type coreConfig struct {
+	Command       string `json:"command"`
+	ProviderType  string `json:"provider_type"`
+	PluginName    string `json:"plugin_name"`
+	TelemetryType string `json:"telemetry_type"`
+}
+
+type PluginConfig struct {
+	CoreConfig   coreConfig `json:"core_config"`
+	PluginConfig json.RawMessage
+}
+
+func validateGenericPluginSpec(c coreConfig) error {
+	if c.Command == "" {
+		return errors.New("generic config invalid: must provide command path")
+	}
+
+	if c.PluginName == "" {
+		return errors.New("generic config invalid: must provide plugin name")
+	}
+
+	if c.TelemetryType == "" {
+		return errors.New("generic config invalid: must provide telemetry type")
+	}
+
+	return nil
+}
+
+type connProvider interface {
+	ClientConn() grpc.ClientConnInterface
+}
+
+func (d *Delegate) newServicesGenericPlugin(
+	ctx context.Context,
+	lggr logger.SugaredLogger,
+	jb job.Job,
+	bootstrapPeers []commontypes.BootstrapperLocator,
+	kb ocr2key.KeyBundle,
+	ocrDB *db,
+	lc ocrtypes.LocalConfig,
+	ocrLogger commontypes.Logger,
+	relayGetter RelayGetter,
+	registrar plugins.RegistrarConfig,
+	pipelineRunner pipeline.Runner,
+	endpointGen telemetry.MonitoringEndpointGenerator,
+) (srvs []job.ServiceCtx, err error) {
+	spec := jb.OCR2OracleSpec
+
+	p := PluginConfig{}
+	err = json.Unmarshal(spec.PluginConfig.Bytes(), &p)
+	if err != nil {
+		return nil, err
+	}
+	cconf := p.CoreConfig
+
+	err = validateGenericPluginSpec(cconf)
+	if err != nil {
+		return nil, err
+	}
+
+	rid, err := spec.RelayID()
+	if err != nil {
+		return nil, fmt.Errorf("%s services: %w: %w", cconf.PluginName, ErrJobSpecNoRelayer, err)
```

Comment:
I see this is pre-existing, but this style strikes me as a little odd. Would it be more fitting for there to be a type instead of a sentinel?
```go
type ErrJobSpecNoRelayer struct {
	PluginName string
	Err error
}

func () Unwrap() { return e.Err}

func () Error() string { return fmt.Sprintf("%s services: OCR2 job spec could not get relayer id: %v", e.PluginName, err) }
```
Then callers can just do:
```suggestion
		return nil, ErrJobSpecNoRelayer{Name:cconf.PluginName, Err:err}
```
The `Is` behavior should be unchanged, and then `As` would allow for checking the `PluginName` without parsing.

Reply:
Could do something similar below, with `failed to get relay %s is it enabled?`.

Reply:
Yep this makes sense: I've added this and also the similar ErrRelayNotEnabled.